java-checks/src/main/java/org/sonar/java/checks/ReleaseSensorsCheck.java

```diff
@@ -55,17 +60,34 @@ enum AcquireReleaseSensor {
       this.acquireMethodMatcher = MethodMatchers.create().ofTypes(sensorClass).names(acquireMethod).withAnyParameters().build();
       this.releaseMethodMatcher = MethodMatchers.create().ofTypes(sensorClass).names(releaseMethod).withAnyParameters().build();
     }
+
+    AcquireReleaseSensor(String acquireSensorClass, String acquireMethod, String releaseSensorClass, String releaseMethod) {
+      this.acquireMethodMatcher = MethodMatchers.create().ofTypes(acquireSensorClass).names(acquireMethod).withAnyParameters().build();
+      this.releaseMethodMatcher = MethodMatchers.create().ofTypes(releaseSensorClass).names(releaseMethod).withAnyParameters().build();
+    }
   }
 
   private static class AcquireReleaseStatus {
     List<Tree> acquireInvocations = new LinkedList<>();
     boolean released = false;
   }
 
+  public static final String ACQUIRE = "acquire";
   public static final String RELEASE = "release";
```

Comment:
Since we only use them here and wouldn't expect other rules to use constants from this rule, I would make them private.

Reply:
good one! The auto-extract is tricky XD